### PR TITLE
feat: Documentation for Fetch API

### PR DESCRIPTION
**Pull request by bharath31**
Added some code samples for Fetch API usage 

https://appsmith-docs-git-fetchapi-get-appsmith.vercel.app/core-concepts/writing-code/javascript-promises#fetch-api

### CODE REVIEW

**Comment by bharath31** [comment]
@arunvjn could you please review my code examples?

File: website/docs/core-concepts/writing-code/javascript-promises.md [hunk 1]
```diff
@@ -212,3 +212,44 @@ In the preceding example:
...
+
+
+## Fetch API
+The Fetch API provides an interface for executing network calls programmatically. You can use `fetch()` to programmatically configure and execute a REST API.  
+
+**Examples**
+
+**GET request**
+```javascript
+await fetch("https://opentdb.com/api.php?amount=10")
```
Comment (arunvjn) [inline website/docs/core-concepts/writing-code/javascript-promises.md:224]: I think this should be
`const questions = await fetch("https://opentdb.com/api.php?amount=10")`

File: website/docs/core-concepts/writing-code/javascript-promises.md [hunk 2]
```diff
@@ -212,3 +212,44 @@ In the preceding example:
...
+	return questions.json()
+```
+
+**POST request**
+```javascript
+await fetch("https://63772c9a5c477765121615ba.mockapi.io/users", {
```
Comment (arunvjn) [inline website/docs/core-concepts/writing-code/javascript-promises.md:230]: Although right, I would probably lose the `await` keyword here. 

Another recommendation is to do something as follows
```javascript
try {
 const response = await fetch("...");
 console.log("Success:", response.json());
} catch(error) {
 console.error("Error:", error);
}
```

**Comment by danciaclara** [comment]
@bharath31 Is this a new framework function. If yes, could you create an issue on docs to link to the engineering issue so that we have more context and also when this is due to be released?

**Comment by bharath31** [comment]
@danciaclara this is already live. Support for fetch API was released as a part of Custom JS Libs

**Review by arunvjn — APPROVED** [review]

**Review by danciaclara — APPROVED** [review]

**Pull request merged**